src/synapse.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Mock implementation of the main Synapse class
+ */
+
+import type { Synapse as ISynapse, SynapseOptions, StorageOptions, TokenAmount } from './types.js'
```

Comment:
I find it easier to reason about types if they're defined in the same file as the class that uses them, vs throwing all types into one central type file.